pkg/frontend/routine.go

```diff
@@ -130,6 +133,16 @@ func (rt *Routine) getConnectionID() uint32 {
 	return rt.getProtocol().ConnectionID()
 }
 
+func (rt *Routine) updateGoroutineId() {
```

Comment:
Is this for debugging/logging only?

Reply:
For analyzing the hung. The hung happens sometimes. The go trace  with the the routineid logged can decide which the go routine is hung and which sql the go routine is running.